spec/new_integration/code_loading/loading_from_slice_spec.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+RSpec.describe "Code loading / Loading from slice directory", :application_integration do
+  before :context do
+    with_directory(@dir = make_tmp_directory) do
+      write "config/application.rb", <<~'RUBY'
+        require "hanami"
+
+        module TestApp
+          class Application < Hanami::Application
+          end
+        end
+      RUBY
+
+      write "slices/main/config/settings.rb", <<~'RUBY'
+        module Main
+          class Settings < Hanami::Application::Settings
```

Comment:
@timriley shouldn't this be just `Hanami::Settings`? as per https://github.com/hanami/hanami/pull/1172

Reply:
@jodosha Just rebased now. And I think I used the old class purely from habit; the updated version of this spec now has this inside `config/settings.rb`:

```ruby
write "slices/main/config/settings.rb", <<~'RUBY'
  raise "This file should never be loaded"
RUBY
```

This should hopefully express the intent of this test much more clearly — it's not really about the contents of this file, it's about the fact that it should _never be autoloaded_ by virtue of its location. Hopefully that `raise` makes it a lot clearer :)